pool/object.go

```diff
@@ -42,17 +81,57 @@ func (p *objectPool) Init(alloc Allocator) {
 func (p *objectPool) Get() interface{} {
 	select {
 	case v := <-p.values:
+		p.trySetGauges()
```

Comment:
I tried that, however that would mean the gauge would never appear as zero because it would be before the select.  I could try using a local var perhaps and call this after assignment, let me try that.